examples/ft_transformer_text.py

```diff
@@ -51,21 +52,147 @@ def __call__(self, sentences: List[str]) -> Tensor:
+        'distilbert-base-uncased',
+        'sentence-transformers/all-distilroberta-v1',
+    ])
+parser.add_argument('--pooling', type=str, default='mean',
+                    choices=['mean', 'cls'])
 args = parser.parse_args()
 
+if args.lora and not args.finetune:
+    raise ValueError('Please also specify finetune when '
+                     'choosing LoRA finetuning.')
+
+
+class PretrainedTextEncoder:
+    def __init__(self, model: str, pooling: str, device: torch.device):
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
+        self.model = AutoModel.from_pretrained(model).to(device)
+        self.device = device
+        self.pooling = pooling
+
+    def __call__(self, sentences: List[str]) -> Tensor:
+        inputs = self.tokenizer(sentences, truncation=True,
+                                padding='max_length', return_tensors='pt')
+        for key in inputs:
+            if isinstance(inputs[key], Tensor):
+                inputs[key] = inputs[key].to(self.device)
+        out = self.model(**inputs)
+        mask = inputs['attention_mask']
+        if self.pooling == 'mean':
+            return mean_pooling(out.last_hidden_state.detach(),
+                                mask).squeeze(1).cpu()
+        elif self.pooling == 'cls':
+            return out.last_hidden_state[:, 0, :].detach().cpu()
+        else:
+            raise ValueError(f'{self.pooling} is not supported.')
+
+
+class TextTokenizer:
+    def __init__(self, model: str):
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
+
+    def __call__(self, sentences: List[str]) -> List[Dict[str, Tensor]]:
+        res = []
+        for s in sentences:
+            inputs = self.tokenizer(s, truncation=True, padding=False,
+                                    return_tensors='pt')
+            res.append(inputs)
+        return res
+
+
+class TextEncoder(torch.nn.Module):
+    def __init__(self, model: str, pooling: str = 'mean', lora: bool = False):
+        super().__init__()
+        self.model = AutoModel.from_pretrained(model)
+
+        if lora:
+            if model == 'distilbert-base-uncased':
+                target_modules = ['ffn.lin1']
+            elif model == 'sentence-transformers/all-distilroberta-v1':
+                target_modules = ['intermediate.dense']
+            else:
+                raise ValueError(f'Model {model} is not specified for '
+                                 f'LoRA finetuning.')
+
+            peft_config = LoraConfig(
+                task_type=TaskType.FEATURE_EXTRACTION,
+                r=32,
+                lora_alpha=32,
+                inference_mode=False,
+                lora_dropout=0.1,
+                bias='none',
+                target_modules=target_modules,
+            )
+            self.model = get_peft_model(self.model, peft_config)
+        self.pooling = pooling
+
+    def forward(self, feat: TensorData) -> Tensor:
+        # [batch_size, num_cols, out_channels]
```

Comment:
What does this comment mean? shouldn't it be `[batch_size, num_cols, max_seq_len]`?

Reply:
Right, it's not `out_channels`, but should be `batch_max_seq_len` as we pad to max length for each batch.